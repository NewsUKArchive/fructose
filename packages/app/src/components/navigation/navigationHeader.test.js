/* globals describe beforeEach it expect */

import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import NavigationHeader from "./navigationHeader";

configure({
  adapter: new Adapter()
});

describe("Navigation Header", () => {
  let app;

  it("Renders without back icon", () => {
    app = shallow(
      <NavigationHeader
        navigateToCallback={() => {}}
        isParentMenu={() => true}
      />
    );
    expect(app).toMatchSnapshot();
  });

  it("Renders with back icon", () => {
    app = shallow(
      <NavigationHeader
        navigateToCallback={() => {}}
        isParentMenu={() => false}
      />
    );
    expect(app).toMatchSnapshot();
  });
});
