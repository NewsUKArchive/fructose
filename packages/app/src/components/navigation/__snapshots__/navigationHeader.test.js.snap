// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Header Renders with back icon 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "skyblue",
        "flexDirection": "row",
        "paddingLeft": 17,
        "paddingTop": NaN,
        "paddingVertical": 28,
      }
    }
  >
    <Component
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Component List
    </Text>
  </Component>
  <MenuSeparator />
</TouchableOpacity>
`;

exports[`Navigation Header Renders without back icon 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "skyblue",
        "flexDirection": "row",
        "paddingLeft": 17,
        "paddingTop": NaN,
        "paddingVertical": 28,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../packages/app/src/components/navigation/back-icon.png",
        }
      }
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "gray",
          "fontSize": 20,
          "textAlign": "center",
        }
      }
    >
      Component List
    </Text>
  </Component>
  <MenuSeparator />
</TouchableOpacity>
`;
